### PR TITLE
[A2-846] Add operator conditions to CreateRule API

### PR DESCRIPTION
**Pull request by tylercloke**
### :nut_and_bolt: Description

Add operator (`equals`, `member-of`) to conditions on `CreateRule` API

### :athletic_shoe: Demo Script / Repro Steps

* `start_all_services`
* `rebuild components/authz-service && rebuild components/automate-gateway`
* create a rule: (`hab pkg install -b core/jq-static` for jq)

```
[7][default:/src:0]# curl -kH "api-token: $TOK" https://localhost/apis/iam/v2beta/projects -d "$(jq -n '{ id: "foo-project", name: "my foo project" }')"
{"project":{"name":"my foo project","id":"foo-project","type":"CUSTOM","projects":["foo-project"]}}
[9][default:/src:0]# curl -kH "api-token: $TOK" https://localhost/apis/iam/v2beta/rules -d "$(jq -n '{ id: "foo-rule", name: "my foo rule", type: "NODE", project_id: "foo-project", conditions: [{ operator: "MEMBER_OF", type: "CHEF_SERVERS", values: ["prod", "staging"]}]}')" | jq .
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   430  100   178  100   252   9368  13263 --:--:-- --:--:-- --:--:-- 22631
{
  "rule": {
    "id": "foo-rule",
    "project_id": "foo-project",
    "name": "my foo rule",
    "type": "NODE",
    "conditions": [
      {
        "type": "CHEF_SERVERS",
        "values": [
          "prod",
          "staging"
        ],
        "operator": "MEMBER_OF"
      }
    ]
  }
}
[10][default:/src:0]# curl -kH "api-token: $TOK" https://localhost/apis/iam/v2beta/rules -d "$(jq -n '{ id: "foo-rule-2", name: "my foo rule", type: "NODE", project_id: "foo-project", conditions: [{ operator: "EQUALS", type: "CHEF_SERVERS", values: ["prod", "staging"]}]}')" | jq .
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   502    0   251  100   251  19307  19307 --:--:-- --:--:-- --:--:-- 38615
{
  "error": "creating rule with ID \"foo-rule-2\": a condition must specify exactly one value for 'equals' operator",
  "message": "creating rule with ID \"foo-rule-2\": a condition must specify exactly one value for 'equals' operator",
  "code": 3,
  "details": []
}
[16][default:/src:0]# curl -kH "api-token: $TOK" https://localhost/apis/iam/v2beta/rules -d "$(jq -n '{ id: "foo-rule-2", name: "my foo rule", type: "NODE", project_id: "foo-project", conditions: [{ operator: "EQUALS", type: "CHEF_SERVERS", values: ["prod"]}]}')" | jq .
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   399  100   167  100   232   5218   7250 --:--:-- --:--:-- --:--:-- 12468
{
  "rule": {
    "id": "foo-rule-2",
    "project_id": "foo-project",
    "name": "my foo rule",
    "type": "NODE",
    "conditions": [
      {
        "type": "CHEF_SERVERS",
        "values": [
          "prod"
        ],
        "operator": "EQUALS"
      }
    ]
  }
}
```

### :white_check_mark: Checklist

- [ ] Necessary tests added/updated?
- [ ] Necessary docs added/updated?
- [ ] Code actually executed?
- [ ] Vetting performed (unit tests, lint, etc.)?

### CODE REVIEW

File: components/authz-service/server/v2/project.go [hunk 1]
```diff
@@ -351,7 +351,12 @@ func storageCondition(ruleType storage.RuleType, apiCondition *api.Condition) (s
 		return storage.Condition{}, err
 	}
 
-	return storage.NewCondition(ruleType, apiCondition.Values, condAttr, storage.MemberOf)
+	condOp, err := fromAPIProjectRuleConditionOperators(apiCondition.Operator)
```
Comment (bcmdarroch) [inline components/authz-service/server/v2/project.go]: shorten to `fromAPIConditionOperators`?

File: components/authz-service/server/v2/project.go [hunk 2]
```diff
@@ -440,6 +452,34 @@ func fromAPIProjectRuleConditionTypes(t api.ProjectRuleConditionTypes) (storage.
...
+}
+
+var apiToStorageConditionOperators = map[api.ProjectRuleConditionOperators]storage.ConditionOperator{}
+var onceReverseConditionOperatorsMapping sync.Once
+
+func fromStorageConditionOperator(t storage.ConditionOperator) (api.ProjectRuleConditionOperators, error) {
+	if s, ok := storageToAPIConditionOperators[t]; ok {
+		return s, nil
+	}
+	return 0, fmt.Errorf("invalid condition operator %s", t.String())
```
Comment (msorens) [inline components/authz-service/server/v2/project.go]: s/%s/%q/
(Helps the consumer to visually set off the entity here.)

File: components/authz-service/server/v2/project.go [hunk 3]
```diff
@@ -440,6 +452,34 @@ func fromAPIProjectRuleConditionTypes(t api.ProjectRuleConditionTypes) (storage.
...
+	onceReverseConditionOperatorsMapping.Do(func() {
+		for k, v := range storageToAPIConditionOperators {
+			apiToStorageConditionOperators[v] = k
+		}
+	})
+
+	if s, ok := apiToStorageConditionOperators[t]; ok {
+		return s, nil
+	}
+	return 0, fmt.Errorf("invalid condition operator %s", t.String())
```
Comment (msorens) [inline components/authz-service/server/v2/project.go]: s/%s/%q/
(Also just noticed that the two lines for "invalid condition type" should have the same change.)

File: components/authz-service/server/v2/project.go (hunk 1 above)
Comment (tylercloke) [reply components/authz-service/server/v2/project.go]: I agree it's a bit verbose but I was following the convention set already by `fromAPIProjectRuleConditionTypes`.

**Pull request merged**